components/web_discovery/browser/BUILD.gn

```diff
@@ -64,3 +93,21 @@ source_set("unit_tests") {
     "//testing/gtest",
   ]
 }
+
+source_set("browser_tests") {
+  testonly = true
+  sources = [ "content_scraper_browsertest.cc" ]
+  deps = [
+    ":browser",
+    "//base/test:test_support",
+    "//brave/components/constants",
+    "//brave/components/web_discovery/common",
+    "//brave/components/web_discovery/common:mojom",
+    "//chrome/test:test_support",
```

Comment:
browser tests are not normally kept in components, b/c you can't depend on //chrome etc (if this is not ContentBrowserTest). The browser test should be moved to src/brave/browser